python/tk_desktop/desktop_window.py

```diff
@@ -1227,82 +1258,48 @@ def __launch_app_proxy_for_project(self, project, requested_pipeline_configurati
-        # fallback.
-        log.debug(
-            "Requested pipeline configuration was not found. Falling back on %s",
-            pprint.pformat(pipeline_configurations[0])
-        )
-        return pipeline_configurations[0]
-
-    def _launch_failed(self, message):
-        self._current_pipeline_descriptor = None
-        message = ("%s"
-                   "\n\nTo resolve this, open Shotgun in your browser\n"
-                   "and check the paths for this Pipeline Configuration."
-                   "\n\nFor more details, see the console." % message)
-        self.project_overlay.show_error_message(message)
-
-    def _sync_success(self, config_path, descriptor):
         try:
-            self._current_pipeline_descriptor = descriptor
-            # Banners might need to be updated, we might have picked a configuration that has been
-            # updated.
-            self._update_banners()
-
-            engine = sgtk.platform.current_engine()
+            self._current_pipeline_descriptor = config_descriptor
+            # Find the interpreter the config wants to use.
+            try:
+                path_to_python = config_descriptor.python_interpreter
+            except TankFileDoesNotExistError:
+                if sys.platform == "darwin":
+                    path_to_python = os.path.join(sys.prefix, "bin", "python")
+                elif sys.platform == "win32":
+                    path_to_python = os.path.join(sys.prefix, "python.exe")
+                else:
+                    path_to_python = os.path.join(sys.prefix, "bin", "python")
 
-            # Signals from PySide return unicode strings, even if the signal is a string,
-            # so re-encode the string as utf8. If we don't well start spreading unicode through
-            # Toolkit, including some environment variables, which will cause issues on Windows.
-            config_path = config_path.encode("utf8")
+            # Create a descriptor for the current core and gets its PYTHONPATH.
+            core_python = sgtk.get_sgtk_module_path()
 
-            # Find where Python is installed so we can launch an interpreter.
-            path_to_python = sgtk.get_python_interpreter_for_config(config_path)
-            core_python = sgtk.get_core_python_path_for_config(config_path)
+            config_path = engine.sgtk.configuration_descriptor.get_path()
 
             # startup server pipe to listen
             engine.startup_rpc()
 
             # pickle up the info needed to bootstrap the project python
             desktop_data = {
                 "core_python_path": core_python,
-                "config_path": config_path,
+                # Every settings that were used for discovering the pipeline configuration must be
+                # passed down to the next process so it can launch the same pipeline.
+                "manager_settings": {
+                    "caching_policy": toolkit_manager.caching_policy,
```

Comment:
would it make sense that rather than picking and choosing parameters here, we added a method 
to toolkit mgr to serialize its state? That way potential new state variables could be handled gracefully? Not sure about this, curious what your thoughts are.